src/main/java/net/neoforged/neoforge/common/world/ModifiableBiomeInfo.java

```diff
@@ -79,6 +86,19 @@ public void applyBiomeModifiers(final Holder<Biome> biome, final List<BiomeModif
             }
         }
         this.modifiedBiomeInfo = builder.build();
+        return !equivalent(originalBiomeInfo, modifiedBiomeInfo, registryAccess);
+    }
+
+    private boolean equivalent(BiomeInfo original, BiomeInfo modified, RegistryAccess registryAccess) {
+        if (!original.climateSettings().equals(modified.climateSettings())) {
+            return false;
+        }
+        var oEffects = original.effects();
+        var mEffects = modified.effects();
+        DynamicOps<JsonElement> ops = RegistryOps.create(JsonOps.INSTANCE, registryAccess);
+        JsonElement oEffectsJson = BiomeSpecialEffects.CODEC.encodeStart(ops, oEffects).result().orElse(null);
+        JsonElement mEffectsJson = BiomeSpecialEffects.CODEC.encodeStart(ops, mEffects).result().orElse(null);
+        return oEffectsJson != null && oEffectsJson.equals(mEffectsJson);
```

Comment:
I don't have anything against this approach, but I think it would be better just to encode everything and compare using the `Biome#NETWORK_CODEC` on the off chance this changes in the future.

Reply:
I can do that, yeah. Should I take the same approach on the `Structure` on the off chance that something nested in that record is no longer `.equals`-comparable somewhere down the line?

Reply:
Oh wait structures aren't even synced... so this logic isn't needed there.